libc/src/setjmp/x86_64/setjmp.cpp

```diff
@@ -11,35 +11,97 @@
 #include "src/__support/macros/config.h"
 #include "src/setjmp/setjmp_impl.h"
 
+#if LIBC_COPT_SETJMP_ENABLE_FORTIFICATION
+#include "src/setjmp/checksum.h"
+#endif
+
 #if !defined(LIBC_TARGET_ARCH_IS_X86_64)
 #error "Invalid file include"
 #endif
 
-namespace LIBC_NAMESPACE_DECL {
+#if LIBC_COPT_SETJMP_ENABLE_FORTIFICATION
+#define ACCUMULATE_CHECKSUM()                                                  \
```

Comment:
There are still two copies of this, one here and the other in `longjmp.cpp`. They have to match, or the checksum computed by `setjmp` won't agree with the one `longjmp` expects. So surely there should be one copy of the implementation, and it should live in a header file that both source files include.

That's also a good place to put the comment explaining the choice of algorithm.

Reply:
fixed. also, it is easy to see there are quality issues in current implementation but I think it is a reasonable compromise for this initial attempt. I have mentioned the issues in the newly added comments and I am open to further improvement if anyone has better ideas. 

```
// For now, the checksum is computed with a simple multiply-xor-rotation
// algorithm. The pesudo code is as follows:
//
// def checksum(x, acc):
//     masked = x ^ MASK
//     high, low = full_multiply(masked, acc)
//     return rotate(high ^ low, ROTATION)
//
// Similar other multiplication-based hashing, zero inputs
// for the `full_multiply` function may pollute the checksum with zero.
// However, user inputs are always masked where the initial ACC amd MASK are
// generated with random entropy and ROTATION is a fixed prime number. It should
// be of a ultra-low chance for masked or acc being zero given a good quality of
// system-level entropy.
```